### PR TITLE
Address management

### PR DESCRIPTION
**Pull request by tanishk15sharma**
- style: address page
- style: address form
- style: responsive layout
- feat: address context
- feat: address reducer
- feat: address state management 
- feat: add address
- feat: edit address
- feat: remove address

### CODE REVIEW

**Review by SukanyaSen530 — APPROVED** [review]
Functionalities working fine. Can be merged!

File: src/components/address/AddressCard.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  };
+  return (
+    <div className="address-card-main">
+      <h2>{name}</h2>
+      <p>
+        {address} {pincode}
+        <span className="sub-txt"> (pincode)</span>
+      </p>
+      <p>
+        {" "}
```
Comment (sushil-mod) [inline src/components/address/AddressCard.js:21]: remove this

File: src/contexts/index.js [hunk 2]
```diff
@@ -3,3 +3,4 @@ export * from "./cart-context";
 export * from "./filter-context";
 export * from "./wishlist-context";
 export * from "./products-context";
+export * from "./address-context";
```
Comment (sushil-mod) [inline src/contexts/index.js:6]: nice you are using barrel exports

File: src/contexts/products-context.js [hunk 3]
```diff
@@ -5,20 +5,23 @@ const productsContext = createContext(null);
...
+  const [productLoading, setProductLoading] = useState(false);
   useEffect(() => {
     (async () => {
       try {
+        setProductLoading(true);
         let response = await axios.get("/api/products");
         setProducts(response.data.products);
+        setProductLoading(false);
       } catch (err) {
         console.log(err);
```
Comment (sushil-mod) [inline src/contexts/products-context.js:17]: remove logs

**Review by sushil-mod — APPROVED** [review]
The clean code and UI look awesome. good to merge

File: src/contexts/index.js (hunk 2 above)
Comment (tanishk15sharma) [reply src/contexts/index.js:6]: Thanks

**Pull request merged**